services/agents/jobs.go

```
@@ -18,60 +18,271 @@
-	req := &agentpb.StartJobRequest{
-		JobId:   jobID,
-		Timeout: durationpb.New(timeout),
-		Job: &agentpb.StartJobRequest_Echo_{
-			Echo: &agentpb.StartJobRequest_Echo{
-				Message: message,
-				Delay:   durationpb.New(delay),
-			},
-		},
+func (s *JobsService) RestartJob(ctx context.Context, jobID string) error {
+	var job *models.Job
+	var artifact *models.Artifact
+	var location *models.BackupLocation
+	var locationConfig *models.BackupLocationConfig
+	var dbConfig *models.DBConfig
+	errTx := s.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		job, err = models.FindJobByID(s.db.Querier, jobID)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		if job.Retries == 0 {
+			return ErrRetriesExhausted
+		}
+
+		job.Retries--
+		if err = tx.Update(job); err != nil {
+			return err
+		}
+
+		switch job.Type {
+		case models.MySQLBackupJob:
+			artifact, err = models.FindArtifactByID(tx.Querier, job.Data.MySQLBackup.ArtifactID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			location, err = models.FindBackupLocationByID(tx.Querier, artifact.LocationID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			dbConfig, err = models.FindDBConfigForService(tx.Querier, job.Data.MySQLBackup.ServiceID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+		case models.MongoDBBackupJob:
+			artifact, err = models.FindArtifactByID(tx.Querier, job.Data.MongoDBBackup.ArtifactID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			location, err = models.FindBackupLocationByID(tx.Querier, artifact.LocationID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			dbConfig, err = models.FindDBConfigForService(tx.Querier, job.Data.MongoDBBackup.ServiceID)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+		case models.MySQLRestoreBackupJob:
+		case models.MongoDBRestoreBackupJob:
```

Hm, maybe we should return an error for that case? Are restores retriable?